packages/addons-v5/commands/addons/wait.js

```diff
@@ -16,20 +17,32 @@ function * run (ctx, api) {
     } else {
       addons = yield api.get('/addons')
     }
-    addons = addons.filter(addon => addon.state === 'provisioning')
   }
+  addons = addons.filter(addon => addon.state === 'provisioning')
 
   let interval = parseInt(ctx.flags['wait-interval'])
   if (!interval || interval < 0) { interval = 5 }
 
   for (let addon of addons) {
-    addon = yield waitForAddonProvisioning(api, addon, interval)
+    const startTime = new Date()
+    try {
+      addon = yield waitForAddonProvisioning(api, addon, interval)
+    } catch (error) {
+      cli.error(error)
+      notify(`heroku addons:wait ${addon.name}`, 'Add-on failed to provision', false)
+      throw error
```

Comment:
Do we need to both `cli.error` and throw the error?

Reply:
no, just `cli.exit(1)`. I changed this behavior in oclif to cause `cli.error()` to actually exit

Reply:
actually even better would be to throw the error like this but remove the `cli.error(error)` line. And even better than that would be to do:

```
try {
  // ...
} finally {
  notify()
}
```

But I'm not sure if finally blocks are allowed in js like this or if this is a ts thing

Reply:
That works in JS.

Reply:
@jdxcode what I'm noticing from my tests is that if I remove `cli.error(error)`, the error message doesn't propagate to the user.  For example the out only looks like this:

`Creating www-redis... !`

when it should be:

```
Creating www-redis... ! The add-on was unable to be created, with status deprovisioned
```

Reply:
@jdxcode Is this what you're proposing?
```    
try {
      addon = yield waitForAddonProvisioning(api, addon, interval)
    } catch (error) {
      notify(`heroku addons:wait ${addon.name}`, 'Add-on failed to provision', false)
    } finally {
      let configVars = (addon.config_vars || [])
      if (configVars.length > 0) {
        configVars = configVars.map(c => cli.color.configVar(c)).join(', ')
        cli.log(`Created ${cli.color.addon(addon.name)} as ${configVars}`)
      }

      // only show notification if addon took longer than 5 seconds to provision
      if (new Date() - startTime >= 1000 * 5) {
        notify(`heroku addons:wait ${addon.name}`, 'Add-on successfully provisioned')
      }
    }
```

Reply:
actually I'm not sure what I was talking about, finally wouldn't work here because it would be called every time. This should work though:

```js
  for (let addon of addons) {
    const startTime = new Date()
    try {
      addon = yield waitForAddonProvisioning(api, addon, interval)
    } catch (err) {
      notify(`heroku addons:wait ${addon.name}`, 'Add-on failed to provision', false)
      throw err
    }

    let configVars = (addon.config_vars || [])
    if (configVars.length > 0) {
      configVars = configVars.map(c => cli.color.configVar(c)).join(', ')
      cli.log(`Created ${cli.color.addon(addon.name)} as ${configVars}`)
    }

    // only show notification if addon took longer than 5 seconds to provision
    if (new Date() - startTime >= 1000 * 5) {
      notify(`heroku addons:wait ${addon.name}`, 'Add-on successfully provisioned')
    }
  }
```

Reply:
@jdxcode Ok, good!  I was confused about that `finally` for the same reason :)